src/main/java/org/dependencytrack/persistence/jdbi/AnalysisDao.java

```diff
@@ -33,4 +42,63 @@ public interface AnalysisDao {
             """)
     void createComments(@Bind List<Long> analysisId, @Bind String commenter, @Bind List<String> comment);
 
+    @SqlQuery("""
+            INSERT INTO "ANALYSISCOMMENT"
+              ("ANALYSIS_ID", "COMMENT", "COMMENTER", "TIMESTAMP")
+            VALUES
+              (:analysisId, :comment, :commenter, NOW())
+            RETURNING *
+            """)
+    @GetGeneratedKeys("*")
+    @RegisterBeanMapper(AnalysisComment.class)
+    AnalysisComment createComment(@Bind long analysisId, String comment, String commenter);
+
+    default AnalysisComment makeAnalysisComment(long analysisId, String comment, String commenter) {
+        if (comment == null) {
+            return null;
+        }
+        return createComment(analysisId, comment, commenter);
+    }
+
+    @SqlQuery("""
+            SELECT EXISTS(
+                SELECT 1 FROM "ANALYSIS"
+                WHERE "PROJECT_ID" = :projectId
+            )
+            """)
+    boolean hasVulnerabilities(@Bind final long projectId);
+
+    @SqlUpdate("""
+            INSERT INTO "ANALYSIS"
+               ("PROJECT_ID", "COMPONENT_ID", "VULNERABILITY_ID", "STATE", "JUSTIFICATION", "RESPONSE", "DETAILS", "SUPPRESSED")
+            VALUES
+               (:projectId, :componentId, :vulnId,
+               <#if state>
+                    :state,
+               <#else>
+                    'NOT_SET',
+               </#if>
```

Comment:
No need for templating here, you can just use [coalesce](https://neon.tech/postgresql/postgresql-tutorial/postgresql-coalesce).

```suggestion
               COALESCE(:state, 'NOT_SET'),
```